### ISSUE TITLE
10.4.x broken on fresh install

### ISSUE THREAD

**Issue by virtuald** [sameersbn/docker-gitlab#1496]
When starting a 10.4.x container with an empty data directory, the container keeps restarting with the following error message:

```
gitlab              | Initializing datadir...
gitlab              | Saving key "/home/git/data/ssh/ssh_host_key" failed: unknown or unsupported key type
gitlab              | Generating OpenSSH host keys... RSA1 Initializing logdir...
gitlab              | Initializing datadir...
```

When using 10.3.x, it successfully generates a key and starts without issues. I'm guessing something changed on the underlying OS?

This appears to be the underlying issue that #1495 is running into.

**Comment by straga** [comment]
ssh-keygen not support rsa1 anymore

**Comment by tlightsky** [comment]
have exact same problem here

**Comment by josefglatz** [comment]
I have the exact same problem even if I don't use volumes.

My host system is Ubuntu 16.04 with latest stable docker. I tried gitlab 10.4.2, 10.4.0 and 10.3.2 without success. Even without any volumes.

It's actually not possible to use this image. Any suggestions for a workaround?

Cheers,
Josef

**Comment by zorgzerg** [comment]
+1

**Comment by edtriccorp** [comment]
@josefglatz I was able to install 10.3.6 and then upgrade to 10.4.1. I do not know if that will work for you but worth a shot. See #1495. Not much technical detail but it started out with 10.4.2 for me which is now running after the long way around.

**Comment by vincentmele** [comment]
modify line 1265 in assets/runtime/functions per https://gotfix.com/docker/gitlab/commit/9da8acc85fb48591f68cd0769080519c19f217f6

Remove the line that has "rsa1" in it.

``` if [[ ! -e ${GITLAB_DATA_DIR}/ssh/ssh_host_rsa_key ]]; then
    echo -n "Generating OpenSSH host keys... "
   # generate_ssh_key rsa1     ${GITLAB_DATA_DIR}/ssh/ssh_host_key
    generate_ssh_key rsa      ${GITLAB_DATA_DIR}/ssh/ssh_host_rsa_key
    generate_ssh_key dsa      ${GITLAB_DATA_DIR}/ssh/ssh_host_dsa_key
    generate_ssh_key ecdsa    ${GITLAB_DATA_DIR}/ssh/ssh_host_ecdsa_key
```

**Comment by solidnerd** [comment]
Hey at all,
something doesn't get catched through my last tests. I prepared a fix for the unspported type it will be available in `10.4.2-1` . Sorry for the disruption.
